pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4035,4 +4035,126 @@ protected void internalTruncateTopic(AsyncResponse asyncResponse, boolean author
             });
         }
     }
+
+    protected void internalSetReplicatedSubscriptionStatus(AsyncResponse asyncResponse, String subName,
+            boolean authoritative, Boolean enabled) {
+        log.info("[{}] Attempting to change replicated subscription status to {} - {} {}", clientAppId(), enabled,
+                topicName, subName);
+
+        if (enabled == null) {
```

Comment:
required = true, it is impossible to be null here, can we use boolean?

Reply:
`enabled` can be null if the request body is empty.
```sh
$ curl -i \
  -X POST \
  -d '' \
  -H 'Content-Type: application/json' \
  http://localhost:8080/admin/v2/persistent/public/default/massakam/subscription/sub/replicatedSubscriptionStatus

HTTP/1.1 400 Bad Request
Date: Thu, 03 Jun 2021 06:04:10 GMT
broker-address: localhost
Content-Type: application/json
Content-Length: 50
Server: Jetty(9.4.39.v20210325)

{"reason":"Boolean type request body is required"}
```
I think the "required = true" specification is just an option used when generating Swagger documentation.
https://docs.swagger.io/swagger-core/v1.5.0/apidocs/io/swagger/annotations/ApiParam.html

Reply:
If boolean is used instead of Boolean, it seems that an error response will be returned unless the request body can be cast to boolean type.
```sh
$ curl -i -X POST -d '' -H 'Content-Type: application/json' http://localhost:8080/admin/v2/persistent/public/default/massakam/subscription/sub1/replicatedSubscriptionStatus

HTTP/1.1 400 Bad Request
Date: Thu, 03 Jun 2021 06:56:00 GMT
broker-address: localhost
Content-Type: application/json
Content-Length: 35
Server: Jetty(9.4.39.v20210325)

The request entity cannot be empty.

$ curl -i -X POST -d '"foo"' -H 'Content-Type: application/json' http://localhost:8080/admin/v2/persistent/public/default/massakam/subscription/sub1/replicatedSubscriptionStatus

HTTP/1.1 400 Bad Request
Date: Thu, 03 Jun 2021 07:01:15 GMT
broker-address: localhost
Content-Type: text/plain
Content-Length: 248
Server: Jetty(9.4.39.v20210325)

Cannot deserialize value of type `boolean` from String "foo": only "true"/"True"/"TRUE" or "false"/"False"/"FALSE" recognized
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 1]

$ curl -i -X POST -d 0 -H 'Content-Type: application/json' http://localhost:8080/admin/v2/persistent/public/default/massakam/subscription/sub1/replicatedSubscriptionStatus

HTTP/1.1 204 No Content
Date: Thu, 03 Jun 2021 06:56:21 GMT
broker-address: localhost
Server: Jetty(9.4.39.v20210325)

$ curl -i -X POST -d 1 -H 'Content-Type: application/json' http://localhost:8080/admin/v2/persistent/public/default/massakam/subscription/sub1/replicatedSubscriptionStatus

HTTP/1.1 204 No Content
Date: Thu, 03 Jun 2021 06:56:34 GMT
broker-address: localhost
Server: Jetty(9.4.39.v20210325)
```
Since this behavior is not a problem, I changed the type of `enabled` to boolean.